modules/system/classes/MailManager.php

```diff
@@ -128,28 +183,35 @@ protected function addContentToMailerInternal($message, $template, $data, $plain
         $swiftMessage = $message->getSwiftMessage();
 
         if (empty($swiftMessage->getSubject())) {
-            $message->subject(Twig::parse($template->subject, $data));
+            if ($html) {
+                $message->subject(Twig::parse($html->subject, $data));
+            } else if ($text) {
+                $message->subject(Twig::parse($text->subject, $data));
+            }
         }
 
         $data += [
             'subject' => $swiftMessage->getSubject()
         ];
 
-        if (!$plainOnly) {
-            /*
-             * HTML contents
-             */
-            $html = $this->renderTemplate($template, $data);
-
-            $message->setBody($html, 'text/html');
+        /*
+         * HTML content
+         */
+        if ($html) {
+            $message->setBody($this->renderTemplate($html, $data), 'text/html');
+            if (!$text && $html->content_text) {
+                $text = $html;
```

Comment:
Shouldn't this be `$text = $html->content_text`?

Reply:
@LukeTowers no, because `renderTextTemplate()` method will be called with this value and it's going to extract the content_text value.